examples/bond/scripts/run/bond-functions.js

```diff
@@ -0,0 +1,328 @@
+    TMPL_NEW_BOND: newAsaInfo[assetID],
+    TMPL_APPLICATION_ID: appInfo.appID,
+    TMPL_STORE_MANAGER: storeManagerAccount.addr
+  };
+  const dexLsig = await deployer.loadLogic('dex-lsig.py', scInitParam);
+  await deployer.optInAcountToASA(newAsaInfo[assetID], 'bob', {});
+  const groupTx = [
+    // Transfer tokens to dex lsig.
+    {
+      type: types.TransactionType.TransferAsset,
+      sign: types.SignType.SecretKey,
+      fromAccount: buyerAccount,
+      toAccountAddr: dexLsig.address(),
+      amount: 10,
+      assetID: asaInfo.assetIndex,
+      payFlags: { totalFee: 1000 }
+    },
+    // New bond token transfer to buyer's address
+    {
+      type: types.TransactionType.TransferAsset,
+      sign: types.SignType.LogicSignature,
+      fromAccountAddr: dexLsig.address(),
+      lsig: dexLsig,
+      toAccountAddr: buyerAccount.addr,
+      amount: 10,
+      assetID: newAsaInfo[assetID],
+      payFlags: { totalFee: 1000 }
+    },
+    {
+      type: types.TransactionType.TransferAlgo,
+      sign: types.SignType.LogicSignature,
+      fromAccountAddr: dexLsig.address(),
+      lsig: dexLsig,
+      toAccountAddr: buyerAccount.addr,
+      amountMicroAlgos: 1000,
+      payFlags: {}
+    },
+    // call to bond-dapp
+    {
+      type: types.TransactionType.CallNoOpSSC,
+      sign: types.SignType.SecretKey,
+      fromAccount: buyerAccount,
+      appID: appInfo.appID,
+      payFlags: {},
+      appArgs: ['str:redeem_coupon']
+    }
+  ];
+
+  console.log('Redeeming tokens!');
+  await executeTransaction(deployer, groupTx);
+  console.log('Tokens redeemed!');
+}
+
+/**
+ * Create buyback lsig and store it's address in bond-dapp
+ * @param {Deployer object} deployer
+ * @param {Account} storeManagerAccount
+ */
+async function createBuyback (deployer, storeManagerAccount) {
+  const scInitParam = {
+    TMPL_APPLICATION_ID: appInfo.appID,
+    TMPL_STORE_MANAGER: storeManagerAccount.addr,
+    TMPL_BOND: newAsaInfo[assetID]
+  };
+  buybackLsig = await deployer.loadLogic('buyback-lsig.py', scInitParam);
+  await fundAccount(deployer, buybackLsig.address());
+
+  const buybackTx = {
+    type: types.TransactionType.CallNoOpSSC,
+    sign: types.SignType.SecretKey,
+    fromAccount: storeManagerAccount,
+    appID: appInfo.appID,
+    payFlags: {},
+    appArgs: ['str:create_buyback', convert.addressToPk(buybackLsig.address())]
+  };
+
+  // Only store manager can allow opt-in to ASA for lsig
+  const optInTx = [
+    {
+      type: types.TransactionType.TransferAlgo,
+      sign: types.SignType.SecretKey,
+      fromAccount: storeManagerAccount,
+      toAccountAddr: buybackLsig.address(),
+      amountMicroAlgos: 0,
+      payFlags: {}
+    },
+    {
+      type: types.TransactionType.OptInASA,
+      sign: types.SignType.LogicSignature,
+      fromAccountAddr: buybackLsig.address(),
+      lsig: buybackLsig,
+      assetID: newAsaInfo[assetID],
+      payFlags: {}
+    }
+  ];
+  await executeTransaction(deployer, optInTx);
```

Comment:
ditto - it's copy paste almost - let's make a function out of it.

Reply:
ok